internal/services/inference/model.go

```diff
@@ -0,0 +1,231 @@
+package inference
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/scaleway/scaleway-sdk-go/api/inference/v1"
+	"github.com/scaleway/scaleway-sdk-go/scw"
+	"github.com/scaleway/terraform-provider-scaleway/v2/internal/httperrors"
+	"github.com/scaleway/terraform-provider-scaleway/v2/internal/locality/regional"
+	"github.com/scaleway/terraform-provider-scaleway/v2/internal/services/account"
+	"github.com/scaleway/terraform-provider-scaleway/v2/internal/types"
+)
+
+func ResourceModel() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: ResourceModelCreate,
+		ReadContext:   ResourceModelRead,
+		DeleteContext: ResourceModelDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Timeouts: &schema.ResourceTimeout{
+			Default: schema.DefaultTimeout(defaultModelTimeout),
+			Create:  schema.DefaultTimeout(defaultModelTimeout),
+			Update:  schema.DefaultTimeout(defaultModelTimeout),
+			Delete:  schema.DefaultTimeout(defaultModelTimeout),
+		},
+		SchemaVersion: 0,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The name of the model",
+			},
+			"url": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "The URL of the model",
+			},
+			"secret": {
```

Comment:
Is only hugging face supported at the moment? If so it would be helpful to be more clear about it here.